hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -148,85 +62,214 @@ public function setComponentStatuses($components) {
     CRM_Core_DAO::executeQuery($sql);
-    return TRUE;
   }
 
-  /** function to check if newly added managed entity exist
-   *  If exist then create mapping for that managed entity
-   *  @param array $caseTypes values are case type names (e.g. "Appraisal")
+  /**
+   * Create a defined list of relationship types
+   *
    */
-  public function manageCaseTypes($caseTypes) {
-    $caseType = implode("','", $caseTypes);
-    $sql = "SELECT * from civicrm_case_type WHERE name IN ('{$caseType}')";
-    $caseFetch = CRM_Core_DAO::executeQuery($sql);
-    $values = array();
-    while ($caseFetch->fetch()) {
-      $values[] = "('org.civicrm.hrcase', '{$caseFetch->name}', 'CaseType', {$caseFetch->id}, 'never')";
+  public static function createRelationshipTypes() {
+    $toInsert = '';
+    foreach(self::relationshipsTypesList() as $relationship) {
+      $toInsert .= "('{$relationship['name_a_b']}','{$relationship['name_a_b']}','{$relationship['name_b_a']}','{$relationship['name_b_a']}','{$relationship['name_b_a']}','Individual','Individual',NULL,NULL,0,1),";
     }
-    if (!empty($values)) {
-      $manageSql = "INSERT INTO civicrm_managed (module, name, entity_type, entity_id, cleanup) VALUES " . implode(',', $values) ;
-      CRM_Core_DAO::executeQuery($manageSql);
+    $toInsert = rtrim($toInsert, ',');
+    $sql = "INSERT INTO `civicrm_relationship_type`
+            (`name_a_b`, `label_a_b`, `name_b_a`, `label_b_a`, `description`, `contact_type_a`, `contact_type_b`, `contact_sub_type_a`, `contact_sub_type_b`, `is_reserved`, `is_active`)
+            VALUES {$toInsert}";
+    CRM_Core_DAO::executeQuery($sql);
+  }
+
+  /**
+   * Remove a defined list of relationship types
+   *
+   */
+  public static function removeRelationshipTypes() {
+    $toDelete = array_map(function ($ar) {return $ar['name_b_a'];}, self::relationshipsTypesList());
+    $toDelete = "'" . implode("','", $toDelete) . "'";
+    $sql = "DELETE FROM `civicrm_relationship_type` WHERE name_b_a IN ({$toDelete})";
+    CRM_Core_DAO::executeQuery($sql);
+  }
+
+  /**
+   * (Enable/Disable) a defined list of relationship types
+   *
+   * @param int $setActive 0 : disable , 1 : enable
+   */
+  public static function toggleRelationshipTypes($setActive) {
+    $toToggle = array_map(function ($ar) {return $ar['name_b_a'];}, self::relationshipsTypesList());
+    $toToggle = "'" . implode("','", $toToggle) . "'";
+    $sql = "UPDATE `civicrm_relationship_type` SET is_active={$setActive} WHERE name_b_a IN ({$toToggle})";
+    CRM_Core_DAO::executeQuery($sql);
+  }
+
+  /**
+   * Remove the case types which created by CiviCRM.
+   *
+   */
+  public static function removeCivicrmCaseTypes() {
+    $toDelete = self::civicrmCaseTypesList();
+    $toDelete = "'" . implode("','", $toDelete) . "'";
+    $sql = "DELETE FROM `civicrm_case_type` WHERE name IN  ({$toDelete})";
```

Comment:
I think this is a little bit better/clearer:
```php
$toDelete = implode("','", $toDelete);
$sql = "DELETE FROM `civicrm_case_type` WHERE name IN  ('{$toDelete}')";
```

What do you think? It applies to all the other methods on this class